plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/AbstractPinotIntegrationSmokeTest.java

```diff
@@ -748,6 +766,70 @@ public void testNonLowerCaseColumnNames()
+        for (int i = 0; i < rowCount; i++) {
+            rows.add(format("('string_%s', '%s', '%s')", i, i, initialUpdatedAt.plusMillis(i * 1000).getEpochSecond()));
+        }
+
+        String mixedCaseColumnNamesTableValues = rows.stream().collect(joining(",", "VALUES ", ""));
+
+        // Test segment query all rows
+        assertQuery("SELECT stringcol, longcol, updatedatseconds" +
+                        "  FROM " + MIXED_CASE_TABLE_NAME,
+                mixedCaseColumnNamesTableValues);
+
+        // Test broker query all rows
+        assertQuery("SELECT stringcol, longcol, updatedatseconds" +
+                        "  FROM  \"SELECT updatedatseconds, longcol, stringcol FROM " + MIXED_CASE_TABLE_NAME + "\"",
+                mixedCaseColumnNamesTableValues);
+
+        String singleRowValues = "VALUES (VARCHAR 'string_3', BIGINT '3', BIGINT '" + initialUpdatedAt.plusMillis(3 * 1000).getEpochSecond() + "')";
+
+        // Test segment query single row
+        assertThat(query("SELECT stringcol, longcol, updatedatseconds" +
+                "  FROM " + MIXED_CASE_TABLE_NAME +
+                "  WHERE longcol = 3"))
+                .matches(singleRowValues)
+                .isFullyPushedDown();
+
+        // Test broker query single row
+        assertThat(query("SELECT stringcol, longcol, updatedatseconds" +
+                "  FROM  \"SELECT updatedatseconds, longcol, stringcol FROM " + MIXED_CASE_TABLE_NAME +
+                "\" WHERE longcol = 3"))
+                .matches(singleRowValues)
+                .isFullyPushedDown();
+    }
+
+    @Test
+    public void testAmbiguousTables()
+    {
+        assertQueryFails("SELECT * FROM " + DUPLICATE_TABLE_LOWERCASE, "Ambiguous table names: (" + DUPLICATE_TABLE_LOWERCASE + ", " + DUPLICATE_TABLE_MIXED_CASE + "|" + DUPLICATE_TABLE_MIXED_CASE + ", " + DUPLICATE_TABLE_LOWERCASE + ")");
+        assertQueryFails("SELECT * FROM " + DUPLICATE_TABLE_MIXED_CASE, "Ambiguous table names: (" + DUPLICATE_TABLE_LOWERCASE + ", " + DUPLICATE_TABLE_MIXED_CASE + "|" + DUPLICATE_TABLE_MIXED_CASE + ", " + DUPLICATE_TABLE_LOWERCASE + ")");
+        assertQueryFails("SELECT * FROM \"SELECT * FROM " + DUPLICATE_TABLE_LOWERCASE + "\"", "Ambiguous table names: (" + DUPLICATE_TABLE_LOWERCASE + ", " + DUPLICATE_TABLE_MIXED_CASE + "|" + DUPLICATE_TABLE_MIXED_CASE + ", " + DUPLICATE_TABLE_LOWERCASE + ")");
+        assertQueryFails("SELECT * FROM \"SELECT * FROM " + DUPLICATE_TABLE_MIXED_CASE + "\"", "Ambiguous table names: (" + DUPLICATE_TABLE_LOWERCASE + ", " + DUPLICATE_TABLE_MIXED_CASE + "|" + DUPLICATE_TABLE_MIXED_CASE + ", " + DUPLICATE_TABLE_LOWERCASE + ")");
+        assertQueryFails("SELECT * FROM information_schema.columns", "Ambiguous table names: (" + DUPLICATE_TABLE_LOWERCASE + ", " + DUPLICATE_TABLE_MIXED_CASE + "|" + DUPLICATE_TABLE_MIXED_CASE + ", " + DUPLICATE_TABLE_LOWERCASE + ")");
+    }
+
+    @Test
+    public void testInformationSchema()
```

Comment:
nit: testInformationSchemaForAmbiguousTables?

Or move these assertions into `testNonLowerTable` (like you have for `testAmbiguousTables`)